src/tap/cmd.rs

```diff
@@ -70,20 +80,30 @@ pub async fn cmd(opts: &super::Opts, mut signal_rx: SignalRx) -> exitcode::ExitC
         let stream = res.stream();
     };
 
-    // Loop over the returned results, printing out log events.
-    // NOTE: This will currently ignore notifications. A later `--verbose` option is planned
-    // to include these.
-    // TODO: https://github.com/vectordotdev/vector/issues/6870
+    // Loop over the returned results, printing out tap events.
     loop {
         tokio::select! {
             biased;
             Some(SignalTo::Shutdown | SignalTo::Quit) = signal_rx.recv() => break,
             Some(Some(res)) = stream.next() => {
                 if let Some(d) = res.data {
-                    for event_string in d.output_events_by_component_id_patterns.iter().filter_map(|ev| ev.as_string()) {
-                        #[allow(clippy::print_stdout)]
-                        {
-                            println!("{}", event_string);
+                    for tap_event in d.output_events_by_component_id_patterns.iter() {
+                        match tap_event {
+                            OutputEventsByComponentIdPatternsSubscriptionOutputEventsByComponentIdPatterns::Log(ev) => {
+                                println!("{}", ev.string);
+                            },
+                            OutputEventsByComponentIdPatternsSubscriptionOutputEventsByComponentIdPatterns::Metric(ev) => {
+                                println!("{}", ev.string);
+                            },
+                            OutputEventsByComponentIdPatternsSubscriptionOutputEventsByComponentIdPatterns::EventNotification(ev) => {
+                                if opts.verbose {
+                                    match ev.notification {
+                                        EventNotificationType::MATCHED => println!(r#"[tap] Pattern "{}" successfully matched."#, ev.pattern),
+                                        EventNotificationType::NOT_MATCHED => eprintln!(r#"[tap] Pattern "{}" failed to match: will retry on configuration reload."#, ev.pattern),
+                                        _ => {},
```

Comment:
It seems like we are covering both notification types here. I might leave out the catch-all to prompt us to consider if `vector tap` should handle any additional notification types as we add them.

Reply:
That makes sense -- there's an `EventNotificationType::Other(_)` that I'm not sure the purpose of which I left as a `NoOp`. I think it comes from the magical derives we have around GraphQL types. Let me know if that's an issue cc: @leebenson

Reply:
Yeah, I'd say that's fine. Our internal enum only adds `Matched` and `NotMatched`, so this must be an async-graphql construct.